### PR TITLE
initial commit

### PR DESCRIPTION
**Pull request by charliepoker**

### CODE REVIEW

File: .gitignore [hunk 1]
```diff
@@ -0,0 +1,3 @@
+node_modules
+package-lock.json
+README.md
```
Comment (ozombo) [inline .gitignore:3]: Why is this ignored?

File: config/config.json [hunk 2]
```diff
@@ -0,0 +1,24 @@
+{
+  "development": {
+    "username": "charlie",
+    "password": "",
+    "database": "charlie_bank",
+    "host": "localhost",
+    "dialect": "mysql",
+    "port": "8889"
+  },
+  "test": {
```
Comment (ozombo) [inline config/config.json]: Do you really need a test for this use case?

File: controller/user-controller.js [hunk 3]
```diff
@@ -0,0 +1,146 @@
...
+exports.signup = (req, res) => {
+  // Generate account No
+  const accountNo = () => {
+    let num = "";
+    while (num.length < 10) {
+      num += Math.floor(Math.random() * 10);
+    }
+    return num;
+  };
+  const generateAccountNo = accountNo();
```
Comment (ozombo) [inline controller/user-controller.js]: Why declare another const here when you can easily run accountNumber: accountNo(); ?

File: controller/user-controller.js [hunk 4]
```diff
@@ -0,0 +1,146 @@
...
+
+  //  Generate Otp
+  const generateOtp = () => {
+    let num = "";
+    while (num.length < 4) {
+      num += Math.floor(Math.random() * 4);
+    }
+    return num;
+  };
+  const generateNewOtp = generateOtp();
```
Comment (ozombo) [inline controller/user-controller.js]: Same as in the case above

File: controller/user-controller.js [hunk 5]
```diff
@@ -0,0 +1,146 @@
...
+    password: bcrypt.hashSync(req.body.password, bcrypt.genSaltSync(8), null),
+    accountNumber: generateAccountNo,
+    phoneNumber: req.body.phoneNumber,
+    otp: generateNewOtp,
+  })
+    .then((user) => {
+      res.status(201).json({
+        status: "ok",
+        message: "User registered",
+        data: {
```
Comment (ozombo) [inline controller/user-controller.js:41]: Why are you not returning everything but the password and otp? What happens in the case when we want to integrate welcome emails?

File: controller/user-controller.js [hunk 6]
```diff
@@ -0,0 +1,146 @@
...
+      if (!verified) {
+        return res.status(401).send({ message: "otp is not verified" });
+      }
+    })
+    .then(() => {
+      const payload = {
+        username: req.bodyusername,
+        password: req.body.password,
+      };
+      const options = { expiresIn: 86400, issuer: "http://localhost:5000" };
```
Comment (ozombo) [inline controller/user-controller.js:90]: If you hard code the issuer this way, how do you handle the issuer when you deploy?

File: controller/user-controller.js [hunk 7]
```diff
@@ -0,0 +1,146 @@
...
+      const secret = config.secret;
+      const token = jwt.sign(payload, secret, options);
+      return res
+        .status(200)
+        .json({
+          status: "ok",
+          message: "otp Verified and User signin successful",
```
Comment (ozombo) [inline controller/user-controller.js]: 👀  this is legendary 😂  if to say my phone camera good, an good tweet content

File: controller/user-controller.js [hunk 8]
```diff
@@ -0,0 +1,146 @@
...
+      } else {
+        user
+          .update({
+            isVerified: true,
+          })
+          .then(
+            res.status(200).json({
+              status: "ok",
+              message: "Verification Complete",
+              data: {
```
Comment (ozombo) [inline controller/user-controller.js]: What will be done with the user data being returned? The message alone is fine

File: middleware/verify-signup.js [hunk 9]
```diff
@@ -0,0 +1,35 @@
+const db = require("../models");
+
+
+
+exports.checkDuplicateUsernameOrEmail = (req, res, next) => {
```
Comment (ozombo) [inline middleware/verify-signup.js:3]: How about including phone number checks here?

**Review by ozombo — CHANGES_REQUESTED** [review]

File: config/config.json (hunk 2 above)
Comment (charliepoker) [reply config/config.json]: I have removed the test variables

File: controller/user-controller.js (hunk 3 above)
Comment (charliepoker) [reply controller/user-controller.js]: completely modified the function

File: controller/user-controller.js (hunk 4 above)
Comment (charliepoker) [reply controller/user-controller.js]: completely modified the function

File: controller/user-controller.js (hunk 5 above)
Comment (charliepoker) [reply controller/user-controller.js:41]: I have returned everything needed

File: controller/user-controller.js (hunk 7 above)
Comment (charliepoker) [reply controller/user-controller.js]: e shock you 😁

File: controller/user-controller.js (hunk 8 above)
Comment (charliepoker) [reply controller/user-controller.js]: The returned data has been deleted

File: middleware/verify-signup.js (hunk 9 above)
Comment (charliepoker) [reply middleware/verify-signup.js:3]: A function to check for phone number duplicate has been implemented

File: .gitignore (hunk 1 above)
Comment (charliepoker) [reply .gitignore:3]: Will be taken care of when I am ready to write the README.md

**Pull request merged**